src/posts/xarray-kvikio/index.md

```diff
@@ -0,0 +1,181 @@
+
+With this `ds.load()` will load directly to GPU memory and `ds` will now contain CuPy arrays. At present there are a few limitations:
+
+1. stores cannot be read with consolidated metadata, and
+2. compression is unsupported by the backend.
+
+## Quick demo
+
+First create an example uncompressed dataset to read from
+
+```
+import xarray as xr
+
+store = "./air-temperature.zarr"
+
+airt = xr.tutorial.open_dataset("air_temperature", engine="netcdf4")
+
+for var in airt.variables:
+    airt[var].encoding["compressor"] = None
+airt.to_zarr(store, mode="w", consolidated=True)
+```
+
+Now read
+
+```
+# consolidated must be False
+ds = xr.open_dataset(store, engine="kvikio", consolidated=False)
+ds.air
+```
+
+```
+<xarray.DataArray 'air' (time: 2920, lat: 25, lon: 53)>
+[3869000 values with dtype=float32]
+Coordinates:
+  * lat      (lat) float32 75.0 72.5 70.0 67.5 65.0 ... 25.0 22.5 20.0 17.5 15.0
+  * lon      (lon) float32 200.0 202.5 205.0 207.5 ... 322.5 325.0 327.5 330.0
+  * time     (time) datetime64[ns] 2013-01-01 ... 2014-12-31T18:00:00
+Attributes:
+    GRIB_id:       11
+    GRIB_name:     TMP
+    actual_range:  [185.16000366210938, 322.1000061035156]
+    dataset:       NMC Reanalysis
+    level_desc:    Surface
+    long_name:     4xDaily Air temperature at sigma level 995
+    parent_stat:   Other
+    precision:     2
+    statistic:     Individual Obs
+    units:         degK
+    var_desc:      Air temperature
+```
+
+Note that we get Xarray's lazy backend arrays by default, and that dimension coordinate variables `lat`, `lon`, `time` were read. At this point this looks identical to what we get with a standard `xr.open_dataset(store, engine="zarr")` command.
+
+Now load a small subset
+
+```python
+type(ds["air"].isel(time=0, lat=10).load().data)
+```
+
+```
+cupy._core.core.ndarray
+```
+
+Success!
+
+Xarray integrates [decently well](https://cupy-xarray.readthedocs.io/quickstart.html) with CuPy arrays so you should be able to test out analysis pipelines pretty easily.
+
+## Cool demo
+
+We don't have a cool demo yet but are looking to develop one very soon!
+
+Reach out if you have ideas. We would love to hear from you.
+
+## Summary
+
+We demonstrate integrating the kvikIO library using Xarray's new backend entrypoints. With everything set up, simply adding `engine="kvikio"` enables direct-to-GPU reads from disk or over the network.
+
+## Acknowledgments
+
+My time on this project was funded by NASA-OSTFL 80NSSC22K0345 "Enhancing analysis of NASA data with the open-source Python Xarray Library"
+
+## Appendix I : Step-by-step install instructions
+
+[Wei Ji Leong](https://github.com/weiji14) helpfully [provided steps](https://discourse.pangeo.io/t/favorite-way-to-go-from-netcdf-xarray-to-torch-tf-jax-et-al/2663/2) to get started on your machine:
```

Comment:
Point to correct link at https://github.com/xarray-contrib/cupy-xarray/pull/10#issuecomment-1218374773
```suggestion
[Wei Ji Leong](https://github.com/weiji14) helpfully [provided steps](https://github.com/xarray-contrib/cupy-xarray/pull/10#issuecomment-1218374773) to get started on your machine:
```